synapse/handlers/room_list.py

```diff
@@ -186,6 +186,10 @@ def get_order_for_room(room_id):
         logger.info("After sorting and filtering, %i rooms remain",
                     len(rooms_to_scan))
 
+        # bail if no rooms to work on
+        if len(rooms_to_scan) == 0:
+            defer.returnValue([])
```

Comment:
This function is supposed to return a dict - see https://github.com/matrix-org/synapse/blob/d02e43b15f6b9b24ffe5e0c0d696f8fd71fc8af3/synapse/handlers/room_list.py#L249 for the shape of it.